axelor-stock/src/main/java/com/axelor/apps/stock/service/StockCorrectionServiceImpl.java

```diff
@@ -134,9 +151,10 @@ public StockMove generateStockMove(StockCorrection stockCorrection) throws Axelo
               stockCorrection.getTrackingNumber());
     }
 
-    BigDecimal realQty = stockCorrection.getRealQty();
-    Product product = stockCorrection.getProduct();
-    TrackingNumber trackingNumber = stockCorrection.getTrackingNumber();
+    if (stockLocationLine == null) {
+      stockLocationLine = stockLocationLineService.createLocationLine(toStockLocation, product);
+    }
+
```

Comment:
Please instead call earlier the dedicated methods in StockLocationLineService `getOrCreateStockLocationLine` and `getOrCreateDetailLocationLine` as this current modification will not create the stock location with a tracking number